components/input-group/InputFieldGroup.tsx

```diff
@@ -0,0 +1,90 @@
+import clsx from 'clsx';
+import React, { useId } from 'react';
+import Icon from '../icon/Icon';
+import { EIConTypes } from '../icon/icon.enum';
+import { TIconType } from '../icon/icon.interface';
+import { Label } from '../typography/Label';
+
+export interface IInputFieldGroupProps {
+  id: string;
+  text?: string;
+  placeholder: string;
+  as:
+    | 'text'
+    | 'password'
+    | 'email'
+    | 'number'
+    | 'tel'
+    | 'url'
+    | 'search'
+    | 'file'
+    | 'hidden';
+  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  state: 'default' | 'error' | 'success';
+  required?: boolean;
+  icon?: TIconType;
+  labelText: string;
+  errorMessage?: string;
+}
+
+export const InputFieldGroup = ({
+  text,
+  placeholder,
+  as = 'text',
+  state = 'default',
+  required = false,
+  icon,
+  labelText,
+  errorMessage,
+}: IInputFieldGroupProps) => {
+  const [inputText, setInputText] = React.useState(text ? text : '');
```

Comment:
I would say that we would need to control the text/state/OnChange/errorMessage in the parent component. After the presentation or at the time of the integration into the next app.